src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/ConsiderPassingBaseTypesAsParameters.cs

```diff
@@ -0,0 +1,227 @@
+                MicrosoftCodeQualityAnalyzersResources.ResourceManager,
+                typeof(MicrosoftCodeQualityAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableStandardMessage =
+            new LocalizableResourceString(
+                nameof(MicrosoftCodeQualityAnalyzersResources.ConsiderPassingBaseTypesAsParametersMessage),
+                MicrosoftCodeQualityAnalyzersResources.ResourceManager,
+                typeof(MicrosoftCodeQualityAnalyzersResources));
+
+        private static readonly LocalizableString s_localizableDescription =
+            new LocalizableResourceString(
+                nameof(MicrosoftCodeQualityAnalyzersResources.ConsiderPassingBaseTypesAsParametersDescription),
+                MicrosoftCodeQualityAnalyzersResources.ResourceManager,
+                typeof(MicrosoftCodeQualityAnalyzersResources));
+
+        internal static readonly DiagnosticDescriptor Rule =
+            DiagnosticDescriptorHelper.Create(
+                RuleId,
+                s_localizableTitle,
+                s_localizableStandardMessage,
+                DiagnosticCategory.Design,
+                RuleLevel.IdeHidden_BulkConfigurable,
+                description: s_localizableDescription,
+                isPortedFxCopRule: true,
+                isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(context =>
+            {
+                var eventArgsTypeSymbol = context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemEventArgs);
+
+                context.RegisterOperationBlockStartAction(context =>
+                {
+                    if (context.OwningSymbol is not IMethodSymbol methodSymbol ||
+                        methodSymbol.IsOverride ||
+                        methodSymbol.Parameters.Length == 0 ||
+                        methodSymbol.IsImplementationOfAnyInterfaceMember() ||
+                        methodSymbol.HasEventHandlerSignature(eventArgsTypeSymbol) ||
+                        !context.Options.MatchesConfiguredVisibility(Rule, methodSymbol, context.Compilation,
+                            defaultRequiredVisibility: SymbolVisibilityGroup.All) ||
+                        context.Options.IsConfiguredToSkipAnalysis(Rule, methodSymbol,
+                            context.Compilation))
+                    {
+                        return;
+                    }
+
+                    var typesUsedPerParameter = PooledConcurrentDictionary<IParameterSymbol, PooledConcurrentSet<ITypeSymbol>>.GetInstance(SymbolEqualityComparer.Default);
```

Comment:
I am absolutely not sure that's the right type to use, feedback welcomed.

Reply:
I wasn't, it's now fixed. If you do a global search in the codebase, it seems that we are not doing it consistently or even not always at all.